Hw8 encapsulation

KutsosMaksym opened the pull request:
Encapsulations of Inner Bank classes

KutsosMaksym committed:
HW7
Changing NullPointer to ArithmeticException while division by 0.

KutsosMaksym committed:
HW8 Student and Bank

KutsosMaksym committed:
Merge remote-tracking branch 'origin/HW8' into HW8

# Conflicts:
#	src/main/java/HW7/Calculator.java

KutsosMaksym committed:
HW8 Student and Bank
Encapsulation

Nikots commented on src/main/java/HWS/HW8/Bank.java:
```diff
@@ -0,0 +1,106 @@
+                System.out.println("Sum of replenishment should be more than 0");
+                return;
+            }
+            Iterator<Account> iterator = accountsList.iterator();
+            while (iterator.hasNext()) {
```
Try to use more simple for-each instead of iterator and while
```suggestion
            for (Account account : accountsList) {
```

Nikots commented on src/main/java/HWS/HW8/Bank.java:
```diff
@@ -0,0 +1,106 @@
+                System.out.println("Sum of withdraw should be more than 0");
+                return;
+            }
+            Iterator<Account> iterator = accountsList.iterator();
+            while (iterator.hasNext()) {
+                Account account = iterator.next();
```
Same as above
```suggestion
         for (Account account : accountsList) {
```

Nikots commented on src/test/java/HWS/HW8/BankTest.java:
```diff
@@ -0,0 +1,68 @@
+    @Test
+    public void checkAccountLimit() {
+        Bank bank = new Bank(3);
+        Bank.Account account1 = bank.createAccount(12003, 100);
+        Bank.Account account2 = bank.createAccount(12004, 205);
+        Bank.Account account3 = bank.createAccount(12005, 350);
+        Bank.Account account4 = bank.createAccount(12006, 70);
```
No need in variables account1...4

Nikots commented on src/main/java/HWS/HW8/Bank.java:
```diff
@@ -0,0 +1,106 @@
+            System.out.println("Impossible to create new account with number: " + number + ". Account with such number already exists.");
+            return null;
+        }
+        accountsList.add(new Account(number, balance));
+        return accountsList.get(accountsList.size() - 1);
```
Not sure if you need to return Account here, but if so try this:
```suggestion
        Account account = new Account(number, balance)
        accountsList.add(account);
        return account;
```

KutsosMaksym committed:
Merge remote-tracking branch 'origin/main' into HW8Encapsulation

KutsosMaksym committed:
HW8 code review.
- All operations will be via Bank object
- New foreach for Replenishment and Withdraw operations
- New methods for display transactions Log
- New methods for display Account Info
